cli/src/commands/generate/components/instantiate.ts

```diff
@@ -0,0 +1,92 @@
+import { Logger } from 'winston';
+import { initLogger } from '../../helper.js';
+import { SchemaDirectory } from '../schema-directory.js';
+import { appendPath, logRequiredMessage, mergeSchemas, renderPath } from '../util.js';
+import { getConstValue, getPropertyValue } from './property.js';
+
+export function instantiateGenericObject(definition: object, schemaDirectory: SchemaDirectory, objectType: string, path: string[], debug: boolean = false, instantiateAll: boolean = false): object {
+    const logger = initLogger(debug);
+    let fullDefinition = definition;
+    if (definition['$ref']) {
+        const ref = definition['$ref'];
+        const schemaDef = schemaDirectory.getDefinition(ref); 
+
+        fullDefinition = mergeSchemas(schemaDef, definition);
+    }
+    // TODO rework to properly separate 'verbose' from 'debug' level logging
+    // logger.debug('Generating ' + objectType + ' object from ' + JSON.stringify(fullDefinition));
```

Comment:
```suggestion
```

I suggest we remove this, and instead raise an issue to address the general logging across the CLI invoked methods. What are your thoughts?

Reply:
Agreed!